packages/router/router.ts

```diff
@@ -327,6 +343,7 @@ export interface RouterInit {
   routes: AgnosticRouteObject[];
   history: History;
   hydrationData?: HydrationState;
+  hasErrorBoundary?: HasErrorBoundaryFunction;
```

Comment:
When looking back over the code with fresh eyes, I was confused thinking that this `hasErrorBoundary` function was actually a boolean when we pass it around within the router, particularly because it's used when we're resolving lazy routes and don't know the final boolean value yet, which made me think it was a bug at first. Maybe the name should be a verb, like `detectErrorBoundary`?

Reply:
yeah, I agree - this code comment is particularly indicative of the potential confusion 😂 

> we use the provided `hasErrorBoundary` function to set the `hasErrorBoundary` route property

Reply:
Updated in [e5f2aa8](https://github.com/remix-run/react-router/pull/10045/commits/e5f2aa890e33a77bbf582dcdd2b7d9319e2cbd7f)

Reply:
oops- missed one thing, amended to [caed6bf](https://github.com/remix-run/react-router/pull/10045/commits/caed6bf0d143a474dfb7a04b91f22c9689cb0297)